lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -175,6 +176,35 @@ def add_payment_method(post, payment_method, options)
           post[:ccnumber] = payment_method.number
           post[:cvv] = payment_method.verification_value unless empty?(payment_method.verification_value)
           post[:ccexp] = exp_date(payment_method)
+          add_stored_credential(post, options)
```

Comment:
Adding the stored credential any place we would take send a `ccnumber` field (which seems consistent with how things are implemented in Core), but it's not clear to me whether or not the idea of stored credentials applies to tokenized payment methods. Visa does have this to say:

> A stored credential is information (including, but not limited to, an account number or payment token) that is stored by a merchant or its agent, a payment facilitator, or a staged digital wallet operator to process future transactions.

source: https://usa.visa.com/dam/VCOM/global/support-legal/documents/stored-credential-transaction-framework-vbs-10-may-17.pdf

So it sounds like the idea of a token used as a payment method isn't necessarily precluding the use of the stored credential fields. Should we just always add stored credential info, when provided, any time we're adding payment information (and not worry about the type of payment method)? Welcome thoughts on this.

Reply:
I think it's right to add them whenever they are present in options, independent of type of PM. However, I believe they only need to be sent for auths and purchases (and the auth step of any other actions that include one).

Reply:
@curiousepic Doesn't look like the `authorize` method is delegated to from any other methods, so it would just be `authorize` and `purchase` where we add the stored credential options to an outgoing request.

Me moving the `add_stored_credential` call to happen directly from `authorize` and `purchase` (rather `add_payment_method`) sound good?